### DISCUSSION TITLE
What is the difference between importing from `/antd/es` and `antd/lib`?

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#42660] [Q&A] [answered]
As the title asks, what is the difference? We mostly do this to get access to AntD component types that are not exposed via the traditional `antd` package path.

Which should be used?

**Comment by zombieJ** [comment]
We've export type from the antd path. In most case, your bundler will auto handle module import with correct one. For detail difference you can google some articles, e.g: https://konstantin.digital/blog/es-modules-all-you-need-to-know

**Reply by bombillazo** [reply]
Most are exported, but some more specific ones are not, for example:

![image](https://github.com/ant-design/ant-design/assets/44002321/5a6d18ad-10cd-4dc7-a8d5-e03494612ba8)

**Reply by BoyYangzai** [reply]
> ![image](https://user-images.githubusercontent.com/44002321/241223608-5a6d18ad-10cd-4dc7-a8d5-e03494612ba8.png)

more exacter import is for less bundle such as less export default and just import what you will use

**Reply by bombillazo** [reply]
Yes, I just wanted to know if there was any difference or recommendation.

**Reply by BoyYangzai** [reply, marked as answer]
> Yes, I just wanted to know if there was any difference or recommendation.

if your bundle tool supports treeshaking enabled, it now looks for files in the `es` directory by default
so you should choose the first one(`es`)
